File: bank_statement/vcb.py
import datetime


def convert_date(date_str):
    date_obj = datetime.datetime.strptime(date_str, '%d-%m-%Y')
    return date_obj.strftime('%d/%m')


def convert_format(input_file_path):
    with open(input_file_path, 'r') as input_file:
        for line in input_file:

            date = convert_date(line[:10])
            line = line[22:].replace('\n', '')
            descStart = 0
            if line[1] == '-':
                line = line[2:]
                descStart += 2
            components = line.split(' - ')
            if ' ' not in components[0]:
                amount = components[0]
            else:
                amount = components[0].split(' ')[-1]
            desc = line[descStart + len(components[0]):].replace(' - ', '')
            print(f"{date};{desc};{amount}")


if __name__ == "__main__":
    import sys

    if len(sys.argv) != 2:
        print("Usage: python convert_format.py <input_file_path>")
    else:
        input_file_path = sys.argv[1]
        convert_format(input_file_path)

File: credit/vcb.go
package credit

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"time"
)

func convertDate(dateStr string) string {
	dateObj, _ := time.Parse("02-01-2006", dateStr)
	return dateObj.Format("02/01")
}

func ConvertVCB(filePath string) (string, error) {
	file, err := os.Open(filePath)
	if err != nil {
		return "", err
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var outputText []string
	for scanner.Scan() {
		line := scanner.Text()
		fmt.Println("converting line: " + line)
		date := convertDate(line[:10])
		line = line[22:]
		descStart := 0
		if line[1] == '-' {
			line = line[2:]
			descStart += 2
		}
		components := strings.Split(line, " - ")
		var amount string
		if strings.Index(components[0], " ") == -1 {
			amount = strings.ReplaceAll(components[0], ",00", "")
		} else {
			amountComponents := strings.Split(components[0], " ")
			amount = amountComponents[len(amountComponents)-1]
		}

		amount = strings.ReplaceAll(amount[:len(amount)-3], ".", ",")
		desc := line[descStart+len(components[0]):]
		desc = strings.ReplaceAll(desc, " - ", "")

		outputText = append(outputText, fmt.Sprintf("%s;%s;%s", date, desc, amount))
	}

	if err := scanner.Err(); err != nil {
		return "", err
	}

	return strings.Join(outputText, "\n"), nil
}
